### ISSUE TITLE
Linear Algebra is not actually parallel

### ISSUE THREAD

**Issue by biona001** [OpenMendel/SnpArrays.jl#118]
Starting Julia with 8 threads, the following code never reaches over 100% CPU usage when monitoring with `htop`
```julia
Threads.nthreads() # returns 8

using Revise, SnpArrays, LinearAlgebra
x = SnpArray(SnpArrays.datadir("EUR_subset.bed")) # 379 by 54051
A = SnpLinAlg{Float64}(x, center=true, scale=true, impute=true)
B = rand(size(x, 1), 1800)
C = zeros(54051, 1800)
mul!(C, Transpose(A), B)

# check answer
truth = Transpose(convert(Matrix{Float64}, x, center=true, scale=true, impute=true))*B
all(isapprox.(truth, C, atol=1e-10)) # true
```
However, if I replace [our custom-defined macros](https://github.com/OpenMendel/SnpArrays.jl/blob/master/src/linalg_direct.jl#L70) with `Threads.@spawn` and `@sync`, I get 800% CPU usage. Is there a reason we don't use the standard macros?

**Comment by biona001** [comment]
Actually, the current code is multithreaded. Sorry for the confusion.

**Comment by biona001** [comment]
I am unfortunately still having this problem with SnpArrays v0.3.14. 
`SnpLinAlg` *sometimes* does not run in parallel, but not sure why. Re-opening this issue.

**Comment by kose-y** [comment]
resolved by #119.
